[Yash Raj Khandelwal] iP

yxshrk opened the pull request:
# MollyBot

> "Your mind should be free for big ideas, not mundane tasks." – **Anonymous**

**MollyBot** helps you manage your to-dos, deadlines, and events effortlessly. It's:

- text-based
- simple to use
- **_incredibly_ fast**

All you need to do is:

1. [Download it from here](https://github.com/yxshrk/ip).
2. Double-click the file.
3. Add your tasks, deadlines, and events.
4. Let **MollyBot** handle the rest! 😉 

And it’s **FREE**!

### Features:
- [x] Managing tasks
- [x] Managing deadlines
- [x] Managing events
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public static void main(String[] args) throws MollyException {
    new Molly("./data/Molly.txt").assistUser();
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

yxshrk committed:
Level 0. Rename, Greet, Exit

yxshrk committed:
Level 1. Echo

yxshrk committed:
Level 2. Add, List

yxshrk committed:
Level 3. Mark as Done

yxshrk committed:
Level 4. ToDos, Events, Deadlines

yxshrk committed:
Automated Text UI Testing

yxshrk committed:
Level 5. Handle Errors

yxshrk committed:
Level 6. Delete

yxshrk committed:
Add javadoc comments

yxshrk committed:
Pushing fixes to bot

yxshrk committed:
Committing Error Handling

yxshrk committed:
Level 7. Save

yxshrk committed:
Making some changes

yxshrk committed:
Merge branch 'branch-Level-7' into 'master'

yxshrk committed:
Making fixes to parsing

yxshrk committed:
Finished Level 8. Dates and Times

yxshrk committed:
Merge branch 'branch-Level-8' into 'master'

yxshrk committed:
Making fixes

yxshrk committed:
Added more OOP

yxshrk committed:
Merge branch 'A-MoreOOP' into 'master'

yxshrk committed:
Add packages

yxshrk committed:
Merge branch 'branch-A-Packages' into 'master'

yxshrk committed:
Merge branch 'add-gradle-supports' into 'master'

yxshrk committed:
Update gradle configuration

yxshrk committed:
Add JUnit tests

yxshrk committed:
Merge branch 'branch-A-JUnit' into 'master'

yxshrk committed:
Fix issues with bot

yxshrk committed:
Fix user input validation

yxshrk committed:
Fix number issues with user input checking

yxshrk committed:
Update JavaDoc comments

yxshrk committed:
Merge branch 'branch-A-JavaDoc' into 'master'

yxshrk committed:
Follow coding standard

yxshrk committed:
Merge branch 'branch-A-CodingStandard' into 'master'

yxshrk committed:
Add level 9 find functionality

yxshrk committed:
Merge branch 'branch-Level-9' into 'master'

yxshrk committed:
Add CheckStyle

yxshrk committed:
Merge branch 'branch-A-CheckStyle' into 'master'

engyuhan commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,138 @@
+import tasklist.TaskList;
+import ui.Ui;
+import mollyexception.MollyException;
+import task.*;
```
Should you be using a wildcard import here?

kevin-pek replied:
Reminder that according to the coding standard, imported classes should be listed explicitly. I saw that you had correctly made explicit imports from the `task` package in some other files, so am pointing it out here in case you missed it.

engyuhan commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,138 @@
+     * @throws MollyException
+     */
+
+    public void handleUserInput(String userInput) throws MollyException {
```
Any reason why you wrote a 100 line function instead of extracting out smaller functions?

engyuhan reviewed (commented):
LGTM, just a few nits to pick.

kevin-pek commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,114 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+import tasklist.TaskList;
+import task.*;
```
Same coding standard violation as previously mentioned.

kevin-pek commented on src/main/java/tasklist/TaskList.java:
```diff
@@ -0,0 +1,105 @@
+package tasklist;
+
+import java.util.ArrayList;
+import task.*;
```
Same coding standard violation as previously mentioned.

kevin-pek commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,138 @@
+                        storage.saveTasks(taskList);
+                    } else {
+                        Ui.printLine();
+                        throw new MollyException("Sorry, invalid format for deadline. Please follow the format: deadline (description) /by (end date). The end date can be in the format DD-MM-YYYY HHmm (24 hour format).");
```
The coding standard recommends keeping the line length shorter than 110 characters, and has a hard limit of 120 characters. For long messages like this, it might be better to do line wrapping to improve readability.
```suggestion
                        throw new MollyException("Sorry, invalid format for deadline. "
                                + "Please follow the format: deadline (description) /by (end date). "
                                + "The end date can be in the format DD-MM-YYYY HHmm (24 hour format).");
```

kevin-pek reviewed (commented):
Your code is well organised and well documented, nice work! Just have a few minor issues related to the coding standard that you might want to consider.

yxshrk committed:
Add GUI for Bot

yxshrk committed:
Merge branch 'branch-Level-10' into 'master'

Airiinnn commented on src/main/java/mainwindow/MainWindow.java:
```diff
@@ -0,0 +1,80 @@
+package mainwindow;
+
+import javafx.animation.PauseTransition;
+import javafx.application.Platform;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+import javafx.util.Duration;
+import mollyexception.MollyException;
+import ui.DialogBox;
+import ui.Ui;
```
The import order does not follow the style guide. Perhaps you've forgotten to reformat the file or have not configured IntelliJ? I noticed the same issue throughout the codebase.

https://se-education.org/guides/tutorials/intellijCodeStyle.html

Airiinnn commented on src/main/java/mollyexception/MollyException.java:
```diff
@@ -0,0 +1,16 @@
+/**
+ * Throws MollyException whenever necessary.
+ */
+
```
Did you miss this empty blank line between documentation block and class?

Airiinnn commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,134 @@
+import tasklist.TaskList;
+import ui.Ui;
+import mollyexception.MollyException;
+import task.*;
```
As others have mentioned, wildcard imports is in violation of the coding standard in this project. You can configure IntelliJ so it does not auto contract the import statements.

https://se-education.org/guides/tutorials/intellijCodeStyle.html

Airiinnn commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,134 @@
+     * @param userInput
+     * @throws MollyException
+     */
+
```
Another missed blank line

Airiinnn commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,134 @@
+                        mollyResponse.append(taskList.addTask(newEvent));
+                        storage.saveTasks(taskList);
+                    } else {
+                        throw new MollyException("Sorry, invalid format for event. Please follow the format: event (description) /from (start date or time) /to (end date or time). The start and end date/times can be in the format DD-MM-YYYY HHmm (24 hour format)");
```
As others have mentioned, you can consider breaking these long lines up into shorter ones to keep within the 120 length limit.

Airiinnn commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,134 @@
+     * @throws MollyException
+     */
+
+    public String handleUserInput(String userInput) throws MollyException {
```
Maybe consider extracting out the logic for handling each command into their own private methods.

Airiinnn commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,114 @@
+
+    public String FILE_PATH;
+    public ArrayList<Task> botMemory;
+    public Storage(String FILE_PATH) {
```
nit: Might be better if you kept spacing between members consistent throughout the project.

Airiinnn commented on src/main/java/Molly.class:
class files should not be put in source control. You can remove them in a commit before adding them to your .gitignore file. If you're running your program with IntelliJ, they should be automatically put in a separate build folder. Maybe you might have misconfigured something?

Airiinnn reviewed (commented):
Good job so far. A couple of nits and suggestions to adhere to the coding standard, and to remove class files.

yxshrk committed:
Add assertions for detecting possible programming errors

Some parts of the code base assume that certain conditions are true,
which can lead to subtle bugs if those assumptions are wrong.

To improve robustness and detect errors earlier, assertions will be
added to check that these assumptions hold true at runtime.

Let's,
* Add assertions to validate expected conditions
* Ensure that runtime checks are added without affecting performance in
  production

yxshrk committed:
Improve code quality by refractoring and adhering to coding standards

Some parts of the code did not conform to coding standards, which could
affect readability and maintainability.

To ensure consistency and improve the overall code quality, let's apply
refractoring where necessary and adhere to established coding
conventions.

yxshrk committed:
Merge pull request #1 from yxshrk/branch-A-Assertions

Add assertions for detecting possible programming errors

yxshrk committed:
Merge branch 'master' into branch-A-CodeQuality

yxshrk committed:
Merge pull request #2 from yxshrk/branch-A-CodeQuality

Branch a code quality

yxshrk committed:
Add reminder functionality to Molly Bot

yxshrk committed:
Add User Guide for Molly

yxshrk committed:
Fix rendering issues

yxshrk committed:
Make final changes

yxshrk committed:
Change png format

yxshrk committed:
Make final changes to Molly

yxshrk committed:
Make more changes
